client/features/api.tsx

```diff
@@ -1,32 +1,40 @@
 import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react';
-import {Zoop, Fave} from '../../src/types/custom'
-// import { RootState } from '../app/store';
+import {Zoop, Fave, User} from '../../src/types/custom'
+import { RootState } from '../app/store';
 
 // Define a service using a base URL and expected endpoints
 export const api = createApi({
     reducerPath: 'api',
     baseQuery: fetchBaseQuery({
         baseUrl: 'http://localhost:3000/api', 
-        // prepareHeaders: (headers: Headers, { getState }) => {
-        //   const token = (getState() as RootState).auth.token;
-        //   if (token) {
-        //     headers.set("Authorization", `Bearer ${token}`);
-        //   }
-        //   return headers;
-        // },
+        prepareHeaders: (headers: Headers, { getState }) => {
+          const token = (getState() as RootState).auth.token;
+          if (token) {
+            headers.set("Authorization", `Bearer ${token}`);
+          }
+          return headers;
+        },
       }),
-    tagTypes: ['Zoop', 'Fave'],
-    endpoints: (builder) => ({
-      getAllZoops: builder.query<Zoop[], void>({
-        query: () => `/zoops`,
-        providesTags: (result, error, arg) =>
-          result
-            ? [...result.map(({ id }) => ({ type: 'Zoop' as const, id })), 'Zoop']
-            : ['Zoop']
+      tagTypes: ['CurrentUser', 'Zoop', 'Fave'],
+      endpoints: (builder) => ({
+        register: builder.mutation({
+          query: ({ email, username, password }) => ({
+            url: "users/register",
+            method: "POST",
+            body: { email, username, password },
+          }),
+          invalidatesTags: ["CurrentUser"],
+        }),
+        getAllZoops: builder.query<Zoop[], void>({
+          query: () => `/zoops`,
+          providesTags: (result, error, arg) =>
+            result
+              ? [...result.map(({ id }) => ({ type: 'Zoop' as const, id })), 'Zoop']
+              : ['Zoop']
+        }),
```

Comment:
nit: doesn't feel like this needs to be in this branch, since this branch doesn't load the zoop list yet - but it's ok if you want to leave it in and test after merging